tools/clang/lib/SPIRV/ModuleBuilder.cpp

```diff
@@ -474,6 +477,55 @@ uint32_t ModuleBuilder::getFunctionType(uint32_t returnType,
   return typeId;
 }
 
+uint32_t ModuleBuilder::getImageType(uint32_t sampledType, spv::Dim dim,
+                                     bool isArray) {
+  const Type *type = Type::getImage(theContext, sampledType, dim,
+                                    /*depth*/ 0, isArray, /*ms*/ 0,
```

Comment:
why do we set a `depth` of 0 ?
`0 indicates not a depth image`
Also why do we set a MS of 0 ?
`0 indicates single-sampled content`
It's not intuitive for me why these defaults are chosen [but, perhaps that's because of my limited shader dev. experience :) ]